handler/create.go

```diff
@@ -107,9 +141,70 @@ func (h *Handler) CreateSession(rw http.ResponseWriter, r *http.Request) {
 	rw.WriteHeader(http.StatusCreated)
 	_ = json.NewEncoder(rw).Encode(&api.Session{
 		ID:          id,
-		AgentMSH:    fmt.Sprintf("https://%s%s", r.Host, path.Join(h.prefix, "config", fmt.Sprintf("%s:%s", id, sessionToken))),
 		SessionURL:  fmt.Sprintf("https://%s%s", r.Host, path.Join(h.prefix, "session", id)),
+		AgentMSH:    fmt.Sprintf("https://%s%s", r.Host, path.Join(h.prefix, "config", fmt.Sprintf("%s:%s", id, sessionToken))),
+		PairingCode: session.PairingCode,
+		PairingURL:  session.PairingURL,
 		AgentConfig: session.AgentConfig,
 		ExpiresAt:   session.ExpiresAt,
 	})
 }
+
+type Request struct {
+	URL string `json:"url"`
+}
+
+type Response struct {
+	Success     bool   `json:"success"`
+	Code        string `json:"code"`
+	PairingURL  string `json:"pairing_url"`
+	RedirectURL string `json:"redirect_url"`
+}
+
+func (h *Handler) pcPair(ctx context.Context, supName, supAvatar string, session *Session) error {
+	h.log.Debugf("pairing to %s ...", h.pcfg.PairingURL)
+
+	if supName == "" || supAvatar == "" {
+		return ErrMissingSupporterInfo
+	}
+
+	jsonRequest, _ := json.Marshal(&Request{
+		URL: fmt.Sprintf("https://%s%s/pairing", h.pcfg.ServerAddress, h.prefix),
```

Comment:
The server address can be something like 0.0.0.0:8080, this is not a valid url that is can be connected to. You need to use the host of the http request to find the originating host of the request. Similarly it is done when returning the session url in the create session request.
```bash
jm@dyne ~ λ https --verify=no -a admin:admin POST ':8080/session' id==helping-joe ttl==3600 supporter_name==help supporter_avatar==k|jq
{
  "ID": "helping-joe",
  "SessionURL": "https://localhost:8080/session/helping-joe",
  "AgentMSH": "https://localhost:8080/config/helping-joe:VFrx3t6Vo89Cb2BdADGZ",
  "PairingCode": "b6ydr1",
  "PairingURL": "https://pc-assist.app/pairing/b6ydr1",
  "AgentConfig": {
    "ServerID": "27C73371BAD57819ABBC8977C57619C382FCF8C846E0DBCAFA77AB72F6D454AA5A0B8CF41151DCBF24D29B6717B2EEAB",
    "MeshName": "plexus/helping-joe/9bnMI",
    "MeshType": 2,
    "MeshID": "mesh/control/TwcFgmnFJBi9wiNhVcPEuMjZal1SS@z9o5gSvehVVDkTABgjD2@ebyroN6ZNCFfI",
    "MeshIDHex": "0x4F07058269C52418BDC2236155C3C4B8C8D96A5D524BECFDA39812BDE8555439130018230F6F9E6F2AE837A64D0857C8",
    "MeshServer": "wss://localhost:8080/agent/helping-joe:VFrx3t6Vo89Cb2BdADGZ"
  },
  "ExpiresAt": "2021-11-16T21:36:21.926363494+01:00"
}
jm@dyne ~ λ http https://pc-assist.app/pairing/b6ydr1
HTTP/1.1 302 Found
*.js: max-age=31536000
Cache-Control: no-cache, private
Content-Length: 390
Content-Type: text/html; charset=UTF-8
Date: Tue, 16 Nov 2021 19:36:33 GMT
Location: https://0.0.0.0:8080//pairing/b6ydr1
Server: Caddy
Status: 302 Found
```